pkg/apis/resource/install/install.go

```diff
@@ -36,5 +37,8 @@ func Install(scheme *runtime.Scheme) {
 	utilruntime.Must(resource.AddToScheme(scheme))
 	utilruntime.Must(v1alpha3.AddToScheme(scheme))
 	utilruntime.Must(v1beta1.AddToScheme(scheme))
-	utilruntime.Must(scheme.SetVersionPriority(v1beta1.SchemeGroupVersion, v1alpha3.SchemeGroupVersion))
+	utilruntime.Must(v1beta2.AddToScheme(scheme))
+	// We should change the serialization version to v1beta2
+	// for 1.34.
```

Comment:
We can do that, although it won't help much with removal of v1beta1: support for that has to be kept until it's certain that there are no stored objects with that version, which is typically impossible to determine.

Can you create an issue for this?

Reply:
Created https://github.com/kubernetes/kubernetes/issues/129889 and also linked to it in the TODO.

I remember this from a previous API I was working on, and I think the idea was that by not changing the serialization version until one version later, it would make rollbacks less risky. But it might be that I don't remember the full reason for this.

Reply:
You are right. https://kubernetes.io/docs/reference/using-api/deprecation-policy/ describes our case (v1beta1 -> v1beta2). We are at "X+3":  v1beta2 and v1beta1 available, v1beta1 is deprecated, v1beta1 as storage version.